projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_memory_error_handler.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import re
+
+from vdk.internal.core import errors
+
+MEMORY_LIMIT_PATTERN = r"Limit=(\d+\.\d+)\s*([KMGTP]B)"
+UNITS = {"KB": 1024, "MB": 1024**2, "GB": 1024**3, "TB": 1024**4}
+THRESHOLDS = [1.2, 1.5, 2.0, 512]
```

Comment:
Should 512 be part of the thresholds?

Reply:
I proposed it as some kind of huge multiplier instead of doing

```
recovery_cursor.execute("set memory_limit=512GB;")
```

Not sure if it works the way I think it does, though, please use your best judgement. 😄

Reply:
This was a mistake... it's now fixed.